src/System.Numerics.Vectors/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1949,219 +2050,104 @@ public static Matrix4x4 Multiply(Matrix4x4 value1, Matrix4x4 value2)
-            result.M22 = value1.M22 * value2;
-            result.M23 = value1.M23 * value2;
-            result.M24 = value1.M24 * value2;
-            result.M31 = value1.M31 * value2;
-            result.M32 = value1.M32 * value2;
-            result.M33 = value1.M33 * value2;
-            result.M34 = value1.M34 * value2;
-            result.M41 = value1.M41 * value2;
-            result.M42 = value1.M42 * value2;
-            result.M43 = value1.M43 * value2;
-            result.M44 = value1.M44 * value2;
-
-            return result;
+            if (Sse.IsSupported)
+            {
+                var all = Sse.SetAllVector128(value2);
+                Sse.Store(&value1.M11, Sse.Multiply(Sse.LoadVector128(&value1.M11), all));
+                Sse.Store(&value1.M21, Sse.Multiply(Sse.LoadVector128(&value1.M21), all));
+                Sse.Store(&value1.M31, Sse.Multiply(Sse.LoadVector128(&value1.M31), all));
+                Sse.Store(&value1.M41, Sse.Multiply(Sse.LoadVector128(&value1.M41), all));
+                return value1;
+            }
+            else
+            {
+                Matrix4x4 result;
+
+                result.M11 = value1.M11 * value2;
+                result.M12 = value1.M12 * value2;
+                result.M13 = value1.M13 * value2;
+                result.M14 = value1.M14 * value2;
+                result.M21 = value1.M21 * value2;
+                result.M22 = value1.M22 * value2;
+                result.M23 = value1.M23 * value2;
+                result.M24 = value1.M24 * value2;
+                result.M31 = value1.M31 * value2;
+                result.M32 = value1.M32 * value2;
+                result.M33 = value1.M33 * value2;
+                result.M34 = value1.M34 * value2;
+                result.M41 = value1.M41 * value2;
+                result.M42 = value1.M42 * value2;
+                result.M43 = value1.M43 * value2;
+                result.M44 = value1.M44 * value2;
+
+                return result;
+            }
         }
 
         /// <summary>
         /// Returns a new matrix with the negated elements of the given matrix.
         /// </summary>
         /// <param name="value">The source matrix.</param>
         /// <returns>The negated matrix.</returns>
-        public static Matrix4x4 operator -(Matrix4x4 value)
-        {
-            Matrix4x4 m;
-
-            m.M11 = -value.M11;
-            m.M12 = -value.M12;
-            m.M13 = -value.M13;
-            m.M14 = -value.M14;
-            m.M21 = -value.M21;
-            m.M22 = -value.M22;
-            m.M23 = -value.M23;
-            m.M24 = -value.M24;
-            m.M31 = -value.M31;
-            m.M32 = -value.M32;
-            m.M33 = -value.M33;
-            m.M34 = -value.M34;
-            m.M41 = -value.M41;
-            m.M42 = -value.M42;
-            m.M43 = -value.M43;
-            m.M44 = -value.M44;
-
-            return m;
-        }
+        public static Matrix4x4 operator -(Matrix4x4 value) => Negate(value);
```

Comment:
Nit: I think the normal case would be to have the implementation in the operator and have `Negate` call it (more people likely use the operator than call the "friendly name")